iothub/device/src/Transport/Amqp/AmqpTransportHandler.cs

```diff
@@ -75,6 +75,29 @@ private void OnDisconnected()
         public override bool IsUsable => !_disposed;
 
         #region Open-Close
+        public override async Task OpenAsync(TimeoutHelper timeoutHelper)
+        {
+            if (Logging.IsEnabled) Logging.Enter(this, timeoutHelper, $"{nameof(OpenAsync)}");
+            lock (_lock)
+            {
+                if (_disposed)
+                {
+                    return;
+                }
+
+                _closed = false;
+            }
+
+            try
+            {
+                await _amqpUnit.OpenAsync(timeoutHelper.RemainingTime()).ConfigureAwait(false);
```

Comment:
Note on this design: having the times here very small is risky and can cause some very odd behaviors.

Reply:
What behaviors in specific? What would be the ideal approach if we were to change this?

Reply:
OpenAsync with very small timeout may cause rapid retries, etc. 

The right approach would be to properly implement CancellationToken cancellation which would be covering _all_ aspects, including the retry handler, etc.

Reply:
Also, I'm not sure this pattern applied to ReceiveAsync won't cause messages to be lost. Hub will only allow 10 pending messages before it stops sending which means apps may stop receiving messages.